transformations/tests/test_pool_allocator.py

```diff
@@ -757,22 +798,65 @@ def test_pool_allocator_temporaries_kernel_nested(frontend, block_dim, directive
     else:
         definitions = ()
 
-    config = {
-        'default': {
-            'mode': 'idem',
-            'role': 'kernel',
-            'expand': True,
-            'strict': True
-        },
-        'routine': [{
-            'name': 'driver',
-            'role': 'driver',
-            'real_kind': 'jwrb',
-        }]
-    }
-
-    scheduler = Scheduler(paths=[basedir], config=SchedulerConfig.from_dict(config), frontend=frontend,
-                          definitions=definitions)
+    stack_size = "400*nz*nlon"
+
+    if stack_size_config == "dict":
```

Comment:
I think for readability it would be better to separate this into different tests.